src/Admin/AdminExtensionInterface.php

```diff
@@ -165,6 +166,14 @@ public function alterObject(AdminInterface $admin, $object);
      */
     public function getPersistentParameters(AdminInterface $admin);
 
+    /**
+     * Get a chance to configure admin before used.
+     * Use this method to define templates.
```

Comment:
It can be done to configure a lot more than template (you can call any setter here). So I would remove the line.

Reply:
Don't you think we can add some examples ?
It is good DX to explain which hook should be use for which use case, don't you think ?

Reply:
An example/some documentation is fine, indeed.

When I read
> Use this method to define templates.

I understand
> The purpose of this method is only to define templates.

Something like
> For instance, templates needed for the extension can be set here
Seem more clear.

Best would be to add some doc inside the /docs folder of course.

Reply:
Agreed, I first wanted to add a bullet list, but checkstyle wasn't happy about that...
Will drop this line as suggested, will see use cases to be added in documentation.